src/controllers/oauthContoller.ts

```diff
@@ -131,6 +132,44 @@ export const getClaims = async (
   }
 };
 
+export const verifyPKCE = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  // getting cliams from code before revoke authorization code, for id_token
+  const { code } = req.body;
+  if (!code) {
+    next();
+    return;
+  }
+  try {
+    const codeData = await prisma.code.findUnique({
+      where: {
+        code: code
+      },
+      select: {
+        codeChallenge: true
+      }
+    });
+    if (codeData?.codeChallenge) {
```

Comment:
if there is a codeChallenge associated with code(authorization code), and code_verifier is missing in the request return proper error in response.